app/javascript/dashboard/routes/dashboard/settings/inbox/components/CampaignTable.vue

```diff
@@ -0,0 +1,272 @@
+            return (
+              <div>
+                <span class={`status status__${row.enabled}`}>
+                  {row.enabled
+                    ? this.$t('CAMPAIGN.LIST.STATUS.ENABLED')
+                    : this.$t('CAMPAIGN.LIST.STATUS.DISABLED')}
+                </span>
+              </div>
+            );
+          },
+        },
+
+        {
+          field: 'sender',
+          key: 'sender',
+          title: this.$t('CAMPAIGN.LIST.TABLE_HEADER.SENDER'),
+          align: 'left',
+          renderBodyCell: ({ row }) => {
+            if (row.sender)
+              return (
+                <div class="row--user-block">
+                  <Thumbnail
+                    src={row.sender.thumbnail}
+                    size="20px"
+                    username={row.sender.name}
+                    status={row.sender.availability_status}
+                  />
+                  <div>
+                    <h6 class="sub-block-title user-name text-truncate">
+                      {row.sender.name}
+                    </h6>
+                  </div>
+                </div>
+              );
+            return '---';
+          },
+        },
+        {
+          field: 'url',
+          key: 'url',
+          title: this.$t('CAMPAIGN.LIST.TABLE_HEADER.URL'),
+          align: 'left',
+          renderBodyCell: ({ row }) => (
+            <div class="text-truncate">
+              <a
+                target="_blank"
+                rel="noopener noreferrer nofollow"
+                href={row.url}
+                title={row.url}
+              >
+                {row.url}
+              </a>
+            </div>
+          ),
+        },
+        {
+          field: 'timeOnPage',
+          key: 'timeOnPage',
+          title: this.$t('CAMPAIGN.LIST.TABLE_HEADER.TIME_ON_PAGE'),
+          align: 'left',
+        },
+
+        {
+          field: 'buttons',
+          key: 'buttons',
+          title: '',
+          align: 'left',
+          renderBodyCell: () => (
+            <div class="button-wrapper">
+              <WootSubmitButton
+                button-text={this.$t('CAMPAIGN.LIST.BUTTONS.EDIT')}
+                icon-class="ion-edit"
+                button-class="link hollow grey-btn"
+                click="openEditPopup(label)"
+              />
+              <WootSubmitButton
+                button-text={this.$t('CAMPAIGN.LIST.BUTTONS.DELETE')}
+                icon-class="ion-close-circled"
+                button-class="link hollow grey-btn"
+                click="openEditPopup(label)"
+              />
+            </div>
+          ),
+        },
+      ],
+    };
+  },
+  computed: {
+    tableData() {
+      if (this.isLoading) {
+        return [];
+      }
+      return this.campaigns.map((item) => {
+        return {
+          ...item,
+          url: item.trigger_rules.url,
+          timeOnPage:
+            item.trigger_rules.timeOnPage || item.trigger_rules.time_on_page,
```

Comment:
I think this might be a miss from our end, either we use camelCase everywhere and transform them on API call, or use snake_case. Mixing both would be confusing.